### PR TITLE
Allow num parameter in tf.linspace to be int64.

### PR DESCRIPTION
**Pull request by codrut3**
According to the API, tf.linspace is defined for num int32 or int64.
However the C++ kernel only allows int32, even though the op in core/ops/math_ops permits int64 too.
So I slightly changed the kernel to allow int64 too. I also added tests for RangeOp and LinSpaceOp.

The following code shows the issue:

```
import numpy as np
import tensorflow as tf

with tf.device("/cpu:0"):
    step = tf.constant(1, tf.int64)
    x = tf.linspace(10.0, 12.0, step)

with tf.Session() as sess:
   x_np = sess.run(x)
   print(x_np)
```

### CODE REVIEW

File: tensorflow/core/kernels/sequence_ops_test.cc [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+#include "tensorflow/core/kernels/ops_testutil.h"
+#include "tensorflow/core/kernels/ops_util.h"
+#include "tensorflow/core/platform/test.h"
+
+namespace tensorflow {
+namespace {
+
+class RangeOpTest : public OpsTestBase {
+ protected:
+  void MakeOp(DataType variable_ref_type) {
```
Comment (vrv) [inline tensorflow/core/kernels/sequence_ops_test.cc]: Is there a better name than "variable_ref_type" here?  Should this be something like "Index type" ?

File: tensorflow/core/kernels/sequence_ops_test.cc [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                     .Input(FakeInput(variable_ref_type))
+                     .Input(FakeInput(variable_ref_type))
+                     .Finalize(node_def()));
+    TF_ASSERT_OK(InitOp());
+  }
+};
+
+class LinSpaceOpTest : public OpsTestBase {
+ protected:
+  void MakeOp(DataType variable_ref_type, DataType num_type) {
```
Comment (vrv) [inline tensorflow/core/kernels/sequence_ops_test.cc]: How about 'index_type' instead of  'num_type'?

**Comment by vrv** [comment]
Overall looks great, just a minor variable naming nit, and then I can kick off tests :)

**Comment by codrut3** [comment]
I changed the variable names, I hope it's fine now. Please let me know if there is anything else I should do.

**Comment by vrv** [comment]
Great!

@tensorflow-jenkins test this please

**Review by vrv — APPROVED** [review]

**Pull request merged**